pyuvsim/simsetup.py

```diff
@@ -1042,69 +1047,77 @@ def initialize_uvdata_from_params(obs_params):
+        bandwidth=None, freq_array=None, channel_width=None, Ntimes=None,
+        integration_time=None, start_time=None, time_array=None, bls=None,
+        antenna_nums=None, antenna_names=None, polarization_array=None, no_autos=False,
+        redundant_threshold=None, write_files=True, path_out=None, complete=False,
+        **kwargs):
     """
     Setup a UVData object from keywords.
 
-    Args:
-        yaml_filename : str (optional)
-            Specify filename for yaml file to write out.
-            Defaults to obsparam.yaml
-        antenna_layout_filename : str (optional)
-            Path to csv file of antenna positions (see documentation for details).
-            Will not overwrite existing files.
-            Defaults to antenna_layout.csv
-        array_layout : dictionary (required if antenna_layout_filename not given)
-            dictionary, keys are integer antenna numbers, values are len-3 antenna positions
-            in ENU coordinates [meters].
-        antenna_names : list of str (optional)
-            If unset, antenna names are assigned as "%s" % antnum.
-        telescope_location : len-3 tuple
-            Telescope location on Earth in LatLonAlt coordinates [deg, deg, meters]
-        telescope_name : str
-            Name of telescope
-        Nfreqs : int
-            Number of frequency channels
-        start_freq : float
-            Starting frequency [Hz]
-        bandwidth : float
-            Total frequency bandwidth of spectral window [Hz]
-        channel_width: float
-            Frequency channel spacing [Hz]
-        freq_array : ndarray
-            frequency array [Hz], if this is specified, it supersedes Nfreqs, start_freq, bandwidth
-        Ntimes : int
-            Number of integration bins
-        integration_time : float
-            Width of time bins [seconds]
-        start_time : float
-            Time of the first integration bin [Julian Date]
-        time_array : ndarray
-            time array [Julian Date]. If this is specified it supersedes values of Ntimes, start_time and integration_time
-        bls : list
-            List of antenna-pair tuples for baseline selection
-        redundant_threshold: float
-            Redundant baseline selection tolerance for selection [meters]
-        antenna_nums : list
-            List of antenna numbers to keep in array
-        polarizations : list
-            List of polarization strings to insert into object
-        no_autos : bool
-            If True, eliminate all auto correlations
-        write_files : bool
-            If True, write out the parameter information to yaml files.
-        path_out : str (optional)
-            Path in which to place generated configuration files, if write_files is True.
-            Defaults to current directory.
-        kwargs : dictionary
-            Any additional valid UVData attribute to assign to object.
-
-    Returns:
-        UVData object with zeroed data_array
+    Parameters
```

Comment:
I'm realizing now that this docstring is missing descriptions for the following:

- write_files : bool (default False) : Write out configuration files to generate the same UVData object as the given keywords.
- path_out : str (optional)   Path to write out config files, if used. Defaults to current working directory.
- complete: bool (default False)  Makes full UVData object with all required parameters. [Should probably default to True, for the likely use cases of this function].
- no_autos : bool (default False) Exclude autocorrelations from baselines.

Reply:
I'm not quite sure what you mean -- all these do have descriptions (pretty close to how you are describing them). Are you suggesting updating the descriptions, or were the descriptions totally hidden for you?

I agree that _usually_ people will want to "complete" the object when using this function, however, the reason I left the default as False was for two reasons: 1) to keep the same behaviour as previously by default (though since the function is so new, it may not matter), and 2) because you can always complete the object afterwards, but its more annoying to _undo_ to the completion. 

I'm still happy to change the default if you think it's better.

Reply:
Oh sorry... I didn't see those keywords in the docstring. Never mind then!

That's a good point on the difficulty in reversing the operation. We can keep the default to False.